text/metrics/0131-otlp-export-behavior.md

```diff
@@ -0,0 +1,43 @@
+
+## Explanation
+
+In order to support Prometheus backends using cumulative values as well as other backends that use delta values, the SDK  needs to be configurable and support an OTLP exporter which handles both cumulative values by default and delta values for export. The implication is that the OTLP metric protocol should support both cumulative and delta reporting strategies.
+
+Users should be allowed to declare an environment variable or configuration field that determines this setting for OTLP exporters.
+
+## Internal details
+
+OTLP exporters can report using the behavior it needs to the Metrics SDK. The SDK can merge the previous state of metrics with current value and return the appropriate values to the exporter.  
+  
+Configurable export behavior is already coded in the Metrics Processor component in the [Go SDK](https://github.com/open-telemetry/opentelemetry-go/pull/840). However, this functionality is hardcoded today and would need to rewritten to handle user-defined configuration. See the OTLP metrics definition in [PR #193](https://github.com/open-telemetry/opentelemetry-proto/pull/193), which support both export behaviors. 
+
+## Trade-offs and mitigations
+
+ High memory usage: To support cumulative exports, the SDK needs to maintain state for each cumulative metrics. This means users with high-cardinality metrics can experience high memory usage. 
+ 
+The high-cardinality metrics use case could be addressed by adding the metrics aggregation processor in the Collector. This would enable the Collector, when configured as an Agent, to support converting delta OTLP to Cumulative OTLP. This functionality requires a single agent for each metric-generating client so that all delta values of a metric are converted by the same Collector instance. 
+
+## Prior art and alternatives
+
+A discussed solution is to convert deltas to cumulative in the Collector both as an agent and as a standalone service. However, supporting conversion in the Collector when it is a standalone service requires implementation of a routing mechanism across all Collector instances to ensure delta values of the same cumulative metric are aggregated by the same Collector instance. 
+
+## Open questions
+
+As stated in the previous section, delta to cumulative conversion in the Collector is needed to support Prometheus type backends. This may be necessary in the Collector in the future because the Collector may also accept metrics from other sources that report delta values. 
+
+The future implementation for this conversion in the Collector is still under discussion. There is a proposal is to add a [Metric Aggregation Processor](https://github.com/open-telemetry/opentelemetry-collector/issues/1422) in the Collector which recommends a solution.
```

Comment:
I thought the functional support of the collector to support cumulative -> delta was supposed to be a part of this proposal? Is that not the case?

Reply:
I figure it belongs in the SDK specification, which I _promise_ to resume working on this week.

Reply:
Hello, the reason I didn't add it is I thought we haven't concluded how the support for cumulative -> delta would look like if deltas are sent to different collector instances.

Reply:
Hello, we haven't discussed in detail how the collector should support cumulative -> delta, and we thought having the OTLP exporter to be configurable would satisfy different backends. In the case of the collector taking metrics from other sources,  I think both cumulative -> delta and delta -> cumulative needs to be added.  I will add a sentence for that.